auth/services/oauth/interface.go

```diff
@@ -20,15 +20,21 @@ package oauth
 
 import (
 	"context"
+	"github.com/nuts-foundation/go-did/did"
+	"github.com/nuts-foundation/nuts-node/auth/oauth"
 	"github.com/nuts-foundation/nuts-node/auth/services"
 	"net/url"
 )
 
 // RelyingParty implements the OAuth2 relying party role.
 type RelyingParty interface {
-	// RequestAccessToken is called by the local EHR node to request an access token from a remote Nuts node.
-	RequestAccessToken(ctx context.Context, jwtGrantToken string, authServerEndpoint url.URL) (*services.AccessTokenResult, error)
 	CreateJwtGrant(ctx context.Context, request services.CreateJwtGrantRequest) (*services.JwtBearerTokenResult, error)
+
+	// RequestRFC003AccessToken is called by the local EHR node to request an access token from a remote Nuts node using Nuts RFC003.
+	RequestRFC003AccessToken(ctx context.Context, jwtGrantToken string, authServerEndpoint url.URL) (*oauth.TokenResponse, error)
+
+	// RequestRFC021AccessToken is called by the local EHR node to request an access token from a remote Nuts node using Nuts RFC021.
+	RequestRFC021AccessToken(ctx context.Context, requestHolder did.DID, verifier did.DID, scopes string) (*oauth.TokenResponse, error)
```

Comment:
not sure this is the right approach, given the variety of oauh2 and openID specs. But we need some moving around anyways, since the rest (OpenID4VP and VCI) is all over the place now